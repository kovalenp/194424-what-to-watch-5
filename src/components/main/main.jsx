import React from "react";
import {connect} from "react-redux";
import PropTypes from "prop-types";

import Header from "../header/header";
import Footer from "../footer/footer";
import MovieCatalog from "../movie-catalog/movie-catalog";
import {movieProps} from "../../validation/propTypes";
import browserHistory from "../../common/browser-history";
import {appRoute, authStatus} from "../../common/constants";
import MyListButton from "../my-list-button/my-list-button";

const Main = (props) => {

  const {promoMovie, isAuth} = props;

  return (
    <>
      <section className="movie-card">
        <div className="movie-card__bg">
          <img
            src={promoMovie.background_image}
            alt={promoMovie.name}
          />
        </div>

        <h1 className="visually-hidden">WTW</h1>

        <Header />

        <div className="movie-card__wrap">
          <div className="movie-card__info">
            <div className="movie-card__poster">
              <img
                src={promoMovie.poster_image}
                alt={promoMovie.name + ` poster`}
                width="218"
                height="327"
              />
            </div>

            <div className="movie-card__desc">
              <h2 className="movie-card__title">{promoMovie.name || ``}</h2>
              <p className="movie-card__meta">
                <span className="movie-card__genre">{promoMovie.genre || ``}</span>
                <span className="movie-card__year">{promoMovie.released || ``}</span>
              </p>

              <div className="movie-card__buttons">
                <button
                  className="btn btn--play movie-card__button"
                  type="button"
                  onClick={ () => browserHistory.push(appRoute.PLAYER.replace(`:id`, promoMovie.id))}
                >
                  <svg viewBox="0 0 19 19" width="19" height="19">
                    <use xlinkHref="#play-s"></use>
                  </svg>
                  <span>Play</span>
                </button>
                <MyListButton movie={promoMovie} isAuth={isAuth}/>
              </div>
            </div>
          </div>
        </div>
      </section>

      <div className="page-content">
        <MovieCatalog />
        <Footer />
      </div>
    </>
  );
};

Main.propTypes = {
  promoMovie: movieProps,
  isAuth: PropTypes.bool,
};

const mapStateToProps = (state) => {
  return {
    promoMovie: state.MOVIES.current,
    isAuth: state.USER.authentication === authStatus.AUTH
  };
};

export default connect(mapStateToProps, null)(Main);
