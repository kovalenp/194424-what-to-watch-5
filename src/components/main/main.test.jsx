import React from "react";
import renderer from "react-test-renderer";
import {Main} from "./main";
import movie from "../../mocks/movie";

jest.mock(`../header/header`, () => ({
  __esModule: true,
  default: () => () => <p>Header</p>,
}));

jest.mock(`../movie-catalog/movie-catalog`, () => ({
  __esModule: true,
  default: () => () => <p>MovieCatalog</p>,
}));

jest.mock(`../my-list-button/my-list-button`, () => ({
  __esModule: true,
  default: () => () => <button>MyListButton</button>,
}));

describe(`Main meets B21`, () => {

  afterAll(() => {
    jest.clearAllMocks();
  });

  it(`and rendered correctly`, () => {
    const tree = renderer.create((
      <Main promoMovie={movie} isAuth={true}/>
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
