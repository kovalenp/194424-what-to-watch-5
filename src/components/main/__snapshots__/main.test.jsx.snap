// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main meets B21 and rendered correctly 1`] = `
Array [
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="Bronson"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="Bronson poster"
            height="327"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Bronson
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Action
            </span>
            <span
              className="movie-card__year"
            >
              2008
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--play movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlinkHref="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
